pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -383,15 +383,12 @@ public synchronized void removeConsumer(Consumer consumer, boolean isResetCursor
                         if (!isResetCursor) {
                             try {
                                 topic.getManagedLedger().deleteCursor(cursor.getName());
-                                topic.getManagedLedger().removeWaitingCursor(cursor);
                             } catch (InterruptedException | ManagedLedgerException e) {
                                 log.warn("[{}] [{}] Failed to remove non durable cursor", topic.getName(), subName, e);
                             }
                         }
                     }, topic.getBrokerService().pulsar().getExecutor());
                 });
-            } else {
-                topic.getManagedLedger().removeWaitingCursor(cursor);
             }
```

Comment:
I remember this was fixing a non-durable sub issue, need to see the CI result

Reply:
Yes, the test [testAddWaitingCursorsForNonDurable2](https://github.com/apache/pulsar/pull/22454/files#diff-2ec08bef8a943b1a830d9a1eef04f2a4eec2bf0d80692005c59a993137c74035R719) passed